java/src/jmri/jmrit/catalog/PreviewDialog.java

```diff
@@ -446,6 +432,8 @@ private long availableMemory() {
             for (int i = 0; i < memoryTest.size(); i++) {
                 memoryTest.remove(i);
             }
+            System.gc();
```

Comment:
Adding on to the discussion about memory:  There's some code here that discusses low on memory conditions.

Does this code really run out of memory?  If so, is it understood why?  Is there something specifically that it's doing?

There are good ways to handle e.g. optional objects that can automatically be garbage college (quickly and cleanly) when memory gets tight, and automatically reloaded when needed.  So if there is a memory issue here, rather than (re)invent the wheel we should use those. Happy to help with that.